### PR TITLE
Insight tags and description

### PR DESCRIPTION
**Pull request by liyiy**
## Changes

*Please describe.*  
*If this affects the frontend, include screenshots.*  

Related to the work for saved insights, we are now implementing tag/description/name editing for insights, similar to how we do it for dashboards right now.

https://user-images.githubusercontent.com/25164963/128757241-9777361c-6885-4a3c-8a2d-ad5602a7c211.mov

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious
- [ ] New/changed UI is decent on smartphones (viewport width around 360px)

### CODE REVIEW

**Comment by paolodamico** [comment]
PR is marked as draft and has failing tests so not sure if it's ready to review, let me know if that's the case, also if you'd like help fixing tests.

Additionally, I recommend adding a more verbose description, potentially with a couple of screenshots.

**Comment by liyiy** [comment]
Failing test seems unrelated (node modules/babel cache issue?)

**Comment by liyiy** [comment]
the weird white padding in the video has been fixed, this is probably now ready for a review 😊 @paolodamico

File: frontend/src/scenes/insights/InsightsNav.tsx
Comment (paolodamico) [inline frontend/src/scenes/insights/InsightsNav.tsx:1]: As a small piece of feedback, it would help speed the review, if you add a few comments inline describing some of this refactoring changes (e.g. this was moved from Insights verbatim).

File: frontend/src/lib/components/Description/Description.tsx
Comment (paolodamico) [inline frontend/src/lib/components/Description/Description.tsx:2]: This is a component particularly scoped for insights / dashboards, not sure if this is the best location to save it (may lead to confusion). Also, adding a short description on what this component does might be helpful.

File: frontend/src/scenes/dashboard/DashboardItem.tsx
Comment (paolodamico) [inline frontend/src/scenes/dashboard/DashboardItem.tsx:333]: Curious why we're doing this, seems like we're still fetching the latest version from the API, and we might be better off using the latest version

File: frontend/src/scenes/insights/insightLogic.ts
Comment (paolodamico) [inline frontend/src/scenes/insights/insightLogic.ts:158]: Worth adding a comment on what this is used for

File: frontend/src/scenes/insights/insightLogic.ts
Comment (paolodamico) [inline frontend/src/scenes/insights/insightLogic.ts:166]: I'm a bit worried this is adding more complexity to all these insight-related logics. I thought we already had a reducer somewhere to store the current insight object?

**Review by paolodamico — CHANGES_REQUESTED** [review]
Looking good! Left some comments inline and some UI/UX comments below,
- There's a weird behavior when adding a description/tags without adding an insight name (i.e. saving it first). Ideally if you click on description on an unsaved insight we would prefill the insight name with a sensible name, but if we don't have this functionality yet, perhaps not allowing to add description/tags before saving?
- When clicking "Add description", the input name gets focused.
    image
- (From recording above too). You can see there's already a description in there but when outside editing mode, it shows as if the description was empty. Not sure if this is leftover value on edit mode from a previous insight or it's a display error on default mode.
- I'm not a fan of the visuals of this (particularly having the tags and description inputs right on the background, perhaps having them on a card? Can't seem to find the latest designs for this, but I'll let @clarkus comment on this.

**Comment by clarkus** [comment]
The most recent header designs are at https://www.figma.com/file/9yWtngNb1AIuf6KmXaEPJA/App-doodles?node-id=900%3A147311

I'm a bit confused about the "editing mode" here. I know this is how it works on dashboards, but there is supposed to be an entire page dedicated to editing the insight definition. You should already be in an editing state when you start changing these values. My concern is that once we have a bit stricter role-based access to work, you'll see non-authors editing insights. If you see a prominent action to "finish editing" you might be confused about the state of the rest of the insight (query definition and visualization).

**Comment by liyiy** [comment]
How about I feature flag it under saved insights for now, after I update some of the requested changes and styling? It seems like this falls into the flow of the whole insights flow revamp better than as its own standalone feature.

File: frontend/src/scenes/insights/insightLogic.ts
Comment (liyiy) [reply frontend/src/scenes/insights/insightLogic.ts:166]: I agree but I also think it makes sense in there because keeping track of which insight we're currently on does seem like an `insightLogic` concern. However the editing functionality can probably moved out. Also I couldn't find that reducer that stored current insight object when I first looked, do you know where it might be?

Comment (paolodamico) [reply frontend/src/scenes/insights/insightLogic.ts:166]: Got it! Let's clarify then where this is used. Particularly worth noting that when viewing a dashboard or on specific insights (where you use insight-specific logic) you don't use this reducer. Can't find it so I may have dreamt it.

File: frontend/src/scenes/insights/InsightsNav.tsx
Comment (paolodamico) [reply frontend/src/scenes/insights/InsightsNav.tsx:1]: I think it'd be more useful in the PR (as if you were reviewing your own code) as it'll be meaningless after the PR is merged.

**Comment by paolodamico** [comment]
Sounds good! Do tag me if you want another review, but also feel free to merge when you think best, it's feature flagged anyways so we can improve it in the future.

**Comment by liyiy** [comment]
I'm going to merge since this should all be feature flagged - the only failing test is this, which isn't related at all
 
Screen Shot 2021-08-12 at 5 54 08 PM

**Pull request merged**
